psst-gui/src/ui/lyrics.rs

```diff
@@ -73,7 +131,58 @@ fn track_lyrics_widget() -> impl Widget<AppState> {
                     .center()
                     .padding(Insets::uniform_xy(theme::grid(1.0), theme::grid(0.5)))
                     .link()
+                    .active(|c: &Ctx<Arc<CommonCtx>, TrackLines>, _env| {
+                        let base_progress_ms = c.ctx.progress.as_millis() as f64;
+                        let now_ms = SystemTime::now().duration_since(UNIX_EPOCH).unwrap().as_millis() as f64;
+                        let elapsed = if matches!(c.ctx.playback_state, PlaybackState::Playing) {
+                            now_ms - c.ctx.last_update_ms as f64
+                        } else {
+                            0.0
+                        };
+                        let progress_ms = base_progress_ms + elapsed;
+                        let offset = offset_storage().lock().unwrap().unwrap_or(0.0);
+                        let adj_progress = progress_ms + offset;
+                        let start_ms = c.data.start_time_ms.parse::<f64>().unwrap_or(0.0);
+                        let parsed_end = c.data.end_time_ms.parse::<f64>().unwrap_or(0.0);
+                        let end_ms = if parsed_end > start_ms { parsed_end } else { start_ms + 800.0 };
+                        adj_progress >= start_ms && adj_progress < end_ms
+                    })
```

Comment:
I would rather have the lyrics check the active duration of the global song ticker, rather than recreating it. 

We could make the track duration updates faster to facilitate the lyrics switch, but this code is quite involved just to make the lyrics update.